### DISCUSSION TITLE
How to clear the query result in Table control? 如何清空Table控件的查询结果呢？

### DISCUSSION THREAD

**Discussion by horsewww** [dotnetcore/BootstrapBlazor#3203] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

How to clear the contents of the Table control? The Table control doesn't have a method like Clear() or Reset() .
查询之后，如何清空Table控件的内容，恢复到无数据的状态？Table 控件没有 Clear 或者 Reset之类的方法，能否提供？

### Describe the solution you'd like

提供 Clear 或者 Reset之类的方法。

**Comment by bb-auto** [comment]
@horsewww Thank you for reporting. We will give triage later.

**Reply by ArgoZhang** [reply]
`Table` 就是显示你给的数据而已，你把数据源的数据清空 `Table` 组件自然就达到你想要的效果了

请点击👇 `Mark as answer` 按钮

**Reply by horsewww** [reply]
我是设置了Table的OnQueryAsync回调，在回调中返回数据，然后调用Table的QueryAsync方法开始查询。这种清空下怎么清空查询结果呢？

**Reply by ArgoZhang** [reply]
`OnQueryAsync` 那个返回值 `Items` 给空集合就可以了啊

**Reply by horsewww** [reply]
感谢回复。
OnQueryAsync 返回空集合是可以解决问题。但是OnQueryAsync中是查询数据库或调用web api的，这么改是不是就会变成这样：

```
bool bClear; //表示是否清空表格
private async Task<XXXXX> OnQueryAsync (...)
{
     if(bClear)
     {
              bClear = false
              return 空集合;
     }
     else
     {
            //调用web api或查询数据库，返回数据集合
      }
}
```
需要清空的时候就要执行：

```
bClear = true;
table.QueryAsync();
```

可能问题解决了，但是这样不太优雅，代码太难看。
不知道是否有更好的方法？谢谢！

**Reply by ArgoZhang** [reply, marked as answer]
1. 代码格式交流时你代码能否格式化一下，你就这样随意写别人看起来不舒服，上面代码我帮你格式化一下
 2. 你的需求目前只能这样，还有集中写法还不如这个
 3. 只能 Table 提供一个实例方法 `Clear` 你调用即可

```
Table.Clear();
```
